kong/plugins/ddtrace/handler.lua

```diff
@@ -194,6 +198,9 @@ if subsystem == "http" then
             -- though it is rounded up. This can be more-precisely allocated if necessary
             sampler = new_sampler(math.ceil(conf.initial_samples_per_second / ngx_worker_count), conf.initial_sample_rate)
         end
+        if not header_tags and (conf and conf.header_tags) then
+            header_tags = utils.normalize_header_tags(conf.header_tags)
```

Comment:
It is possible to write a custom validator, however, that's where I draw the line (and Kong's engineer too). A validator tells if an input is valid or not while a normalizer "standardizes" an input.

From the HTTP_HEADER_TAGS RFC, `D!ata__d/o!g` is a valid input but has to be normalized to be usable.
We can argue whether `D!ata__d/o!g` is a valid input or not. I do not think it is, as a user I would certainly not expect my input to be normalized. Worst, I suspect it would likely not result in finding the intended header. 

All in all, there is an RFC and as a product (APM) I rather provide consistent behaviour to our customers than my subjective interpretation.